src/private_cuda.h

```diff
@@ -107,7 +107,8 @@ GPUARRAY_LOCAL int cuda_record(gpudata *, int);
 /* private flags are in the upper 16 bits */
 #define CUDA_WAIT_READ  0x10000
 #define CUDA_WAIT_WRITE 0x20000
-#define CUDA_WAIT_MASK  0x30000
+#define CUDA_WAIT_FORCE 0x40000
```

Comment:
What if an op would request data inputs from 2 device and do strange thinks
like that? Or write output to 2 defice or so some manual transfer?

On Thu, Jun 16, 2016 at 4:05 PM, abergeron <notifications@github.com> wrote:

> In src/private_cuda.h
> <https://github.com/Theano/libgpuarray/pull/196#discussion_r67414529>:
>
> > @@ -107,7 +107,8 @@ GPUARRAY_LOCAL int cuda_record(gpudata *, int);
> >  /* private flags are in the upper 16 bits */
> >  #define CUDA_WAIT_READ  0x10000
> >  #define CUDA_WAIT_WRITE 0x20000
> > -#define CUDA_WAIT_MASK  0x30000
> > +#define CUDA_WAIT_FORCE 0x40000
>
> I don't where and how FORCE would be used in theano. This is only required
> for cross-device synchronization.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/libgpuarray/pull/196/files/66cc5cc095b3713bbbcf9e26798ec72790df1ab1#r67414529>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALC-2NaexdWjWwZ432oeaRI0Y0JZCIuks5qMaxygaJpZM4I3tFE>
> .
>